src/mongoc/mongoc-server-description.h

```diff
@@ -22,6 +22,19 @@
 #include "mongoc-read-prefs.h"
 #include "mongoc-host-list.h"
 
+typedef enum
+{
+   MONGOC_SERVER_UNKNOWN,
+   MONGOC_SERVER_STANDALONE,
+   MONGOC_SERVER_MONGOS,
+   MONGOC_SERVER_POSSIBLE_PRIMARY,
+   MONGOC_SERVER_RS_PRIMARY,
+   MONGOC_SERVER_RS_SECONDARY,
+   MONGOC_SERVER_RS_ARBITER,
+   MONGOC_SERVER_RS_OTHER,
+   MONGOC_SERVER_RS_GHOST,
```

Comment:
I didn't want to define MONGOC_SERVER_DESCRIPTION_TYPES in the public API since it's only used internally; however, I've added a comment here mentioning that the constant should be updated if this enum is ever extended.